src/OpenSpaceToolkit/Astrodynamics/Trajectory/State.cpp

```diff
@@ -499,6 +497,77 @@ State State::inFrame(const Shared<const Frame>& aFrameSPtr) const
     };
 }
 
+std::unordered_map<Shared<const CoordinateSubset>, bool> State::isNear(
+    const State& aState, const std::unordered_map<Shared<const CoordinateSubset>, Real>& aToleranceMap
+) const
+{
+    if (aToleranceMap.size() != coordinatesBrokerSPtr_->accessSubsets().getSize())
+    {
+        throw ostk::core::error::runtime::Wrong("Tolerance Map");
+    }
+
+    const State deltaState = aState - *this;
+
+    std::unordered_map<Shared<const CoordinateSubset>, bool> isNearMap;
+    for (const auto& [subsetSPtr, tolerance] : aToleranceMap)
+    {
+        if (!coordinatesBrokerSPtr_->hasSubset(subsetSPtr))
+        {
+            throw ostk::core::error::runtime::Wrong("Tolerance Map Coordinate Subset");
+        }
+        if (tolerance <= 0.0)
+        {
+            throw ostk::core::error::runtime::Wrong("Tolerance");
+        }
+
+        const VectorXd deltaCoordinates = deltaState.extractCoordinate(subsetSPtr);
+
+        isNearMap[subsetSPtr] = (deltaCoordinates.norm() <= tolerance);
+    }
+    return isNearMap;
+}
+
+std::unordered_map<Shared<const CoordinateSubset>, Array<bool>> State::isNear(
+    const State& aState, const std::unordered_map<Shared<const CoordinateSubset>, VectorXd>& aToleranceArrayMap
+) const
+{
+    if (aToleranceArrayMap.size() != coordinatesBrokerSPtr_->accessSubsets().getSize())
+    {
+        throw ostk::core::error::runtime::Wrong("Tolerance Map");
+    }
+
+    const State deltaState = aState - *this;
+
+    std::unordered_map<Shared<const CoordinateSubset>, Array<bool>> isNearMap;
+    for (const auto& [subsetSPtr, toleranceArray] : aToleranceArrayMap)
+    {
+        if (!coordinatesBrokerSPtr_->hasSubset(subsetSPtr))
+        {
+            throw ostk::core::error::runtime::Wrong("Tolerance Map Coordinate Subset");
+        }
+
+        const VectorXd deltaCoordinates = deltaState.extractCoordinate(subsetSPtr);
+
+        if (toleranceArray.size() != deltaCoordinates.size())
+        {
+            throw ostk::core::error::runtime::Wrong("Tolerance Array Size");
+        }
+        if ((toleranceArray.array() <= 0.0).any())
+        {
+            throw ostk::core::error::runtime::Wrong("Tolerance Array");
+        }
+
+        Array<bool> isNearArray;
+        isNearArray.reserve(deltaCoordinates.size());
+        for (int i = 0; i < deltaCoordinates.size(); ++i)
+        {
+            isNearArray.add(std::abs(deltaCoordinates[i]) <= toleranceArray[i]);
+        }
+        isNearMap[subsetSPtr] = isNearArray;
```

Comment:
```suggestion
        Eigen::ArrayXb isNear = deltaCoordinates.abs() <= toleranceArray
```
Can probably simplify by leveraging eigen here.